src/Build/FileSystem/MSBuildFileSystemAdapter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Shared.FileSystem;
+
+namespace Microsoft.Build.FileSystem
+{
+     public class MSBuildFileSystemAdapter : IFileSystem
```

Comment:
Should this implement IMSBuildFileSystem?

Reply:
MSBuild uses IFileSystem internally (to avoid breaking the archaic yet oldie but goldie nuget version), so this adapter takes an IMSBuildFileSystem and makes it look like an IFileSystem, hence it extends IFileSystem.

Reply:
Why does the adapter class have to be public? Thanks!

Reply:
Good catch! It shouldn't be.